src/EditCriteriaForm.js

```diff
@@ -0,0 +1,146 @@
+import React from "react";
+import { Button, Table } from "semantic-ui-react";
+import CriterionEditRow from "./CriterionEditRow";
+import { analyzeJudgingRoundUrl, criterionOptionSpecPostURL } from "./utils";
+import { withRouter } from "react-router";
+
+class EditCriteriaForm extends React.Component {
+  state = {
+    data: {
+      results: [],
+      rows: []
+    },
+    keys: [],
+    rows: []
+  };
+  constructor(props) {
+    super(props);
+    this.judging_round_id = props.judging_round_id;
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.handleWeightChangeCallback = this.handleWeightChangeCallback.bind(
+      this
+    );
+    this.handleCountChangeCallback = this.handleCountChangeCallback.bind(this);
+  }
+
+  submitFunction(id, count, weight) {
+    let url = criterionOptionSpecPostURL + id + "/";
+    return fetch(url, {
+      credentials: "include",
+      mode: "cors",
+      method: "PATCH",
+      headers: {
+        "Content-Type": "application/json; charset=utf-8"
+        // "Content-Type": "application/x-www-form-urlencoded",
+      },
+      body: JSON.stringify({
+        count: count,
+        weight: weight
+      })
+    });
+  }
+  setRows(data) {
+    let data_rows = data["data"]["results"];
+    var rows = data_rows.reduce(function(obj, x) {
+      obj[x.criterion_option_spec_id] = {
+        weight: x.weight,
+        count: x.count,
+        criterion: x,
+        spec_id: x.criterion_option_spec_id,
+        option: x.criterion_type == "judge" ? x.option : x.criterion_name
+      };
+      return obj;
+    }, {});
+    var keys = [];
+    for (let i = 0; i < data_rows.length; i++) {
+      let spec_id = data_rows[i].criterion_option_spec_id;
+      if (!keys.includes(spec_id)) {
+        keys.push(spec_id);
+      }
+    }
+
+    this.setState({ rows, keys });
```

Comment:
setting the state will cause the component to re-render. im wondering if this is responsible for the infinite loop we are seeing with requests.